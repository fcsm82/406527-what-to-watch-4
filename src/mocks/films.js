import {getRandomElement, generateUniqueId} from "../utils/utils.js";

const FILMS_NUMBER = 8;

const TITLES = [
  `Fantastic Beasts: The Crimes of Grindelwald`,
  `Bohemian Rhapsody`,
  `Macbeth`,
  `Aviator`,
  `We need to talk about Kevin`,
  `What We Do in the Shadows`,
  `Revenant`,
  `Johnny English`,
  `Shutter Island`,
  `Pulp Fiction`,
  `No Country for Old Men`,
  `Snatch`,
  `Moonrise Kingdom`,
  `Seven Years in Tibet`,
  `Midnight Special`,
  `War of the Worlds`,
  `Dardjeeling Limited`,
  `Orlando`,
  `Mindhunter`,
  `Midnight Special`
];

const POSTERS = [
  `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  `bohemian-rhapsody.jpg`,
  `macbeth.jpg`,
  `aviator.jpg`,
  `we-need-to-talk-about-kevin.jpg`,
  `what-we-do-in-the-shadows.jpg`,
  `revenant.jpg`,
  `johnny-english.jpg`,
  `shutter-island.jpg`,
  `pulp-fiction.jpg`,
  `no-country-for-old-men.jpg`,
  `snatch.jpg`,
  `moonrise-kingdom.jpg`,
  `seven-years-in-tibet.jpg`,
  `midnight-special.jpg`,
  `war-of-the-worlds.jpg`,
  `dardjeeling-limited.jpg`,
  `orlando.jpg`,
  `mindhunter.jpg`,
  `midnight-special.jpg`
];

export const TOP_FILM = {
  title: `The Grand Budapest Hotel`,
  poster: `the-grand-budapest-hotel-poster.jpg`,
  genre: `Drama`,
  releaseDate: 2014
};

export const createFilms = () => {
  let films = [];
  for (let i = 0; i < FILMS_NUMBER; i++) {
    films.push({
      id: generateUniqueId(),
      title: getRandomElement(TITLES),
      poster: getRandomElement(POSTERS),
    });
  }
  return films;
};
