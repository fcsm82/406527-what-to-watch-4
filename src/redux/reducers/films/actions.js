export const Action = {
  CHANGE_FILTER: `Change filter`,
  SHOW_MORE: `Show more films`,
};

export const Creator = {
  filterFilms: (filter) => {
    return {
      type: Action.CHANGE_FILTER,
      payload: filter
    };
  },

  showMore: () => {
    return {
      type: Action.SHOW_MORE,
    };
  }
};
