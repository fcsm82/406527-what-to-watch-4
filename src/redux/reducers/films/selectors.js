// import {createSelector} from "reselect";
import NameSpace from "../name-space.js";
// import {getUniqueGenres} from "../../../utils/utils.js";
// import {getFilms} from "../data/selectors.js";

// export const getFilters = createSelector(
//     getFilms,
//     (films) => getUniqueGenres(films)
// );

export const getActiveFilter = (state) => {
  return state[NameSpace.FILMS].activeFilter;
};

export const getShownFilms = (state) => {
  return state[NameSpace.FILMS].shownFilms;
};

export const getView = (state) => {
  return state[NameSpace.FILMS].view;
};
