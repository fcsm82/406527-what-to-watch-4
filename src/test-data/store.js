import NameSpace from "../store/reducers/name-space.js";
import {promoFilm, films} from "../test-data/films.js";
import {user} from "../test-data/user.js";

export const testStore = {
  [NameSpace.DATA]: {
    isAppLoading: true,
    isAppError: false,
    films,
    promoFilm,
    favoriteFilms: [],
    comments: []
  },
  [NameSpace.FILMS]: {
    shownFilms: 8,
    activeFilter: `All genres`,
  },
  [NameSpace.USER]: {
    error: null,
    authorizationStatus: `No auth`,
    authorizationError: false,
    user
  },
};

