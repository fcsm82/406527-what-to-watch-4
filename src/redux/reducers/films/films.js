
import {Action} from "../../actions.js";
// import {View} from "../../../components/app/app.jsx";
import {FILTER} from "../../../consts.js";

// import {getPromoFilm, getFilms} from "../data/selectors.js";

const initialState = {
  // view: View.LIST,
  shownFilms: 8,
  activeFilter: FILTER.ALL,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.CHANGE_FILTER:
      const filter = action.payload;

      return Object.assign({}, state, {
        activeFilter: filter,
        shownFilms: 8
      });

      // case Action.CHANGE_VIEW:
      //   return Object.assign({}, state, {
      //     view: View.DETAILS,
      //     activeFilm: action.payload
      //   });

      // case Action.CHANGE_VIEW_TO_SIGN_IN:
      //   return Object.assign({}, state, {
      //     view: View.SIGN_IN,
      //   });

    case Action.RENDER_FILMS:
      return Object.assign({}, state, {
        shownFilms: state.shownFilms + 8
      });

      // case Action.PLAY_VIDEO:
      //   return Object.assign({}, state, {
      //     view: View.VIDEO
      //   });

      // case Action.EXIT_VIDEO:
      //   return Object.assign({}, state, {
      //     view: View.LIST
      //   });

    default:
      return state;
  }
};

export {reducer};
