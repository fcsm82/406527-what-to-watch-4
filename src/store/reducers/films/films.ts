
import {Action} from "./actions";
import {ALL_GENRES} from "../../../consts";

const FilmsCount = {
  ON_START: 8,
  SHOW_MORE: 8
};

const initialState = {
  shownFilms: FilmsCount.ON_START,
  activeFilter: ALL_GENRES,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case Action.CHANGE_FILTER:
      const filter = action.payload;

      return Object.assign({}, state, {
        activeFilter: filter,
        shownFilms: FilmsCount.ON_START
      });

    case Action.SHOW_MORE:
      return Object.assign({}, state, {
        shownFilms: state.shownFilms + FilmsCount.SHOW_MORE
      });

    default:
      return state;
  }
};

export {reducer};
