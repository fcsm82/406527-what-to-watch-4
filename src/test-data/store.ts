import NameSpace from "../store/reducers/name-space";
import {promoFilm, films} from "../test-data/films";
import {user} from "../test-data/user";

export const testStore = {
  [NameSpace.DATA]: {
    isAppLoading: true,
    isAppError: false,
    films,
    promoFilm,
    favoriteFilms: [],
    comments: []
  },
  [NameSpace.FILMS]: {
    shownFilms: 8,
    activeFilter: `All genres`,
  },
  [NameSpace.USER]: {
    error: null,
    authorizationStatus: `No auth`,
    authorizationError: false,
    user
  },
};

