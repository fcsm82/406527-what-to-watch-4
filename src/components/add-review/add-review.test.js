import React from "react";
import {Router} from "react-router-dom";
import renderer from "react-test-renderer";

import {Provider} from 'react-redux';

import configureStore from 'redux-mock-store';

import AddReview from "./add-review.jsx";
import history from "../../history.js";

import {testStore} from "../../test-data/store.js";
import {film} from "../../test-data/films.js";

const mockStore = configureStore([]);

const store = mockStore(testStore);

it(`Should add review page render correctly`, () => {
  const tree = renderer
    .create(
        <Router history={history}>
          <Provider store={store}>
            <AddReview
              film={film}
              isLoading={false}
              isError={false}
              onSubmitClick={()=>{}}
              onRatingChange={()=>{}}
              onReviewChange={()=>{}}
              isSubmitDisabled={false}
            />
          </Provider>
        </Router>
        , {
          createNodeMock: () => {
            return {};
          }
        }).toJSON();

  expect(tree).toMatchSnapshot();
});
