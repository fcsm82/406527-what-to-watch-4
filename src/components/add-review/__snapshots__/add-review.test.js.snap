// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should add review page render correctly 1`] = `
Array [
  <div
    className="visually-hidden"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
      xmlnsXlink="http://www.w3.org/1999/xlink"
    >
      <symbol
        viewBox="0 0 19 20"
      >
        <title>
          +
        </title>
        <g
          fill="none"
          fillRule="evenodd"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            points="10.778 11.288 10.778 19.553 8.4165 19.553 8.4165 11.288 0.62793 11.288 0.62793 8.9268 8.4165 8.9268 8.4165 0.66211 10.778 0.66211 10.778 8.9268 18.566 8.9268 18.566 11.288"
          />
        </g>
      </symbol>
      <symbol
        fill="#FFF9D9"
        viewBox="0 0 27 27"
      >
        <path
          clipRule="evenodd"
          d="M23.8571 0H16V3.14286H23.8571V11H27V3.14286V0H23.8571Z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M27 23.8571V16H23.8571V23.8571H16V27H23.8571H27L27 23.8571Z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="m0 3.1429v7.8571h3.1429v-7.8571h7.8571v-3.1429h-11v3.1429z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M3.14286 27H11V23.8571H3.14286L3.14286 16H0L0 23.8571V27H3.14286Z"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        viewBox="0 0 18 14"
      >
        <path
          clipRule="evenodd"
          d="M2.40513 5.35353L6.1818 8.90902L15.5807 0L18 2.80485L6.18935 14L0 8.17346L2.40513 5.35353Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        viewBox="0 0 14 21"
      >
        <g
          fill="#EEE5B5"
          fillRule="evenodd"
          strokeWidth="1"
        >
          <polygon
            fillRule="nonzero"
            points="0 -1.1191e-13 4 -1.1191e-13 4 21 0 21"
          />
          <polygon
            fillRule="nonzero"
            points="10 -1.1191e-13 14 -1.1191e-13 14 21 10 21"
          />
        </g>
      </symbol>
      <symbol
        id="play-s"
        viewBox="0 0 19 19"
      >
        <path
          clipRule="evenodd"
          d="M0 0L19 9.5L0 19V0Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "#A6B7AC",
      }
    }
  >
    <div
      className="movie-card__header"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Gangs of new york"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <nav
          className="breadcrumbs"
        >
          <ul
            className="breadcrumbs__list"
          >
            <li
              className="breadcrumbs__item"
            >
              <a
                className="breadcrumbs__link"
                href="/films/1"
                onClick={[Function]}
              >
                Gangs of new york
              </a>
            </li>
            <li
              className="breadcrumbs__item"
            >
              <a
                className="breadcrumbs__link"
              >
                Add review
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="user-block"
        >
          <a
            href="/login"
            onClick={[Function]}
            style={
              Object {
                "color": "inherit",
                "cursor": "pointer",
                "textDecoration": "none",
              }
            }
          >
            Sign In
          </a>
        </div>
      </header>
      <div
        className="movie-card__poster movie-card__poster--small"
      >
        <img
          alt="Gangs of new york"
          height="327"
          src="https://htmlacademy-react-3.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg"
          width="218"
        />
      </div>
    </div>
    <div
      className="add-review"
    >
      <form
        action="#"
        className="add-review__form"
        onSubmit={[Function]}
      >
        <div
          className="rating"
        >
          <div
            className="rating__stars"
            onChange={[Function]}
          >
            <input
              className="rating__input"
              disabled={false}
              id="star-1"
              name="rating"
              type="radio"
              value={1}
            />
            <label
              className="rating__label"
              htmlFor="star-1"
            >
              Rating 
              1
            </label>
            <input
              className="rating__input"
              disabled={false}
              id="star-2"
              name="rating"
              type="radio"
              value={2}
            />
            <label
              className="rating__label"
              htmlFor="star-2"
            >
              Rating 
              2
            </label>
            <input
              className="rating__input"
              disabled={false}
              id="star-3"
              name="rating"
              type="radio"
              value={3}
            />
            <label
              className="rating__label"
              htmlFor="star-3"
            >
              Rating 
              3
            </label>
            <input
              className="rating__input"
              disabled={false}
              id="star-4"
              name="rating"
              type="radio"
              value={4}
            />
            <label
              className="rating__label"
              htmlFor="star-4"
            >
              Rating 
              4
            </label>
            <input
              className="rating__input"
              disabled={false}
              id="star-5"
              name="rating"
              type="radio"
              value={5}
            />
            <label
              className="rating__label"
              htmlFor="star-5"
            >
              Rating 
              5
            </label>
          </div>
        </div>
        <div
          className="add-review__text"
          style={
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0.3)",
            }
          }
        >
          <textarea
            className="add-review__textarea"
            id="review-text"
            name="review-text"
            onChange={[Function]}
            placeholder="Review text"
            required={true}
          />
          <div
            className="add-review__submit"
          >
            <button
              className="add-review__btn"
              disabled={false}
              type="submit"
            >
              Post
            </button>
          </div>
        </div>
      </form>
    </div>
  </section>,
]
`;
